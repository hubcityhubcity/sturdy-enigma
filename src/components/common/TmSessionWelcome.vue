<template>
  <SessionFrame>
    <div id="session-welcome">
      <router-link to="/">
        <img
          class="session-logo-mobile"
          src="~assets/images/lunie-logo-white.svg"
          alt="a small spinning circle to display loading"
        />
      </router-link>

      <h2 class="session-title">
        Welcome to Lunie
      </h2>

      <p class="session-paragraph">
        Lunie is the staking and governance platform for proof-of-stake
        blockchains.
      </p>

      <div class="session-list">
        <LiSession
          id="create-new-address"
          icon="person_add"
          title="Create a new address"
          route="create"
        />
        <LiSession
          id="use-an-existing-address"
          icon="person"
          title="Use an existing address"
          route="existing"
        />
      </div>

      <p class="footnote">
        By using Lunie, you accept our
        <router-link to="/terms" class="link">
          Terms of Service
        </router-link>
        and
        <router-link to="/privacy" class="link"> Privacy Policy </router-link>.
      </p>
    </div>
  </SessionFrame>
</template>

<script>
import SessionFrame from "common/SessionFrame.vue"
import LiSession from "common/TmLiSession"
export default {
  name: `session-welcome`,
  components: {
    SessionFrame,
    LiSession
  }
}
</script>
