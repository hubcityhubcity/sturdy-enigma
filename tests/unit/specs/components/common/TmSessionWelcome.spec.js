import { shallowMount } from "@vue/test-utils"
import TmSessionWelcome from "common/TmSessionWelcome"

describe(`TmSessionWelcome`, () => {
  let $store, wrapper

  beforeEach(() => {
    const state = {
      session: {
        insecureMode: true,
        browserWithLedgerSupport: null
      }
    }
    $store = {
      commit: jest.fn(),
      dispatch: jest.fn(),
      state
    }
    wrapper = shallowMount(TmSessionWelcome, {
      mocks: {
        $store,
        isMobile: function() {
          return false
        }
      },
      stubs: [`router-link`]
    })
  })

  it(`has the expected html structure`, async () => {
    expect(wrapper.element).toMatchSnapshot()
  })

  it(`has the expected html structure when on mobile`, async () => {
    wrapper = shallowMount(TmSessionWelcome, {
      mocks: {
        $store,
        isMobile: function() {
          return true
        }
      },

      stubs: [`router-link`]
    })
    expect(wrapper.element).toMatchSnapshot()
  })
})
