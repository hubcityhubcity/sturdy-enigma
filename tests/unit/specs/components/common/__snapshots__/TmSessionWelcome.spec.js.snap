// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TmSessionWelcome has the expected html structure 1`] = `
<sessionframe-stub>
  <div
    id="session-welcome"
  >
    <router-link-stub
      to="/"
    >
      <img
        alt="a small spinning circle to display loading"
        class="session-logo-mobile"
        src="~assets/images/lunie-logo-white.svg"
      />
    </router-link-stub>
     
    <h2
      class="session-title"
    >
      
      Welcome to Lunie
    
    </h2>
     
    <p
      class="session-paragraph"
    >
      
      Lunie is the staking and governance platform for proof-of-stake
      blockchains.
    
    </p>
     
    <div
      class="session-list"
    >
      <lisession-stub
        icon="person_add"
        id="create-new-address"
        route="create"
        title="Create a new address"
      />
       
      <lisession-stub
        icon="person"
        id="use-an-existing-address"
        route="existing"
        title="Use an existing address"
      />
    </div>
     
    <p
      class="footnote"
    >
      
      By using Lunie, you accept our
      
      <router-link-stub
        class="link"
        to="/terms"
      >
        
        Terms of Service
      
      </router-link-stub>
      
      and
      
      <router-link-stub
        class="link"
        to="/privacy"
      >
         Privacy Policy 
      </router-link-stub>
      .
    
    </p>
  </div>
</sessionframe-stub>
`;

exports[`TmSessionWelcome has the expected html structure when on mobile 1`] = `
<sessionframe-stub>
  <div
    id="session-welcome"
  >
    <router-link-stub
      to="/"
    >
      <img
        alt="a small spinning circle to display loading"
        class="session-logo-mobile"
        src="~assets/images/lunie-logo-white.svg"
      />
    </router-link-stub>
     
    <h2
      class="session-title"
    >
      
      Welcome to Lunie
    
    </h2>
     
    <p
      class="session-paragraph"
    >
      
      Lunie is the staking and governance platform for proof-of-stake
      blockchains.
    
    </p>
     
    <div
      class="session-list"
    >
      <lisession-stub
        icon="person_add"
        id="create-new-address"
        route="create"
        title="Create a new address"
      />
       
      <lisession-stub
        icon="person"
        id="use-an-existing-address"
        route="existing"
        title="Use an existing address"
      />
    </div>
     
    <p
      class="footnote"
    >
      
      By using Lunie, you accept our
      
      <router-link-stub
        class="link"
        to="/terms"
      >
        
        Terms of Service
      
      </router-link-stub>
      
      and
      
      <router-link-stub
        class="link"
        to="/privacy"
      >
         Privacy Policy 
      </router-link-stub>
      .
    
    </p>
  </div>
</sessionframe-stub>
`;
